### PR TITLE
reference: Document multi-compose file support

### PR DESCRIPTION
**Pull request by doanac**
Signed-off-by: Andy Doan <andy@foundries.io>

### CODE REVIEW

**Comment by doanac** [comment]
Docs for 1d186bd8c1d6225b286bd6b19df7b42af4985bbd are browesable at: https://ci.foundries.io/projects/fio-docs/builds/1235/docs/artifacts/html/index.html

File: source/reference-manual/docker/compose-files.rst [hunk 1]
```diff
@@ -0,0 +1,31 @@
+.. _ref-compose-files:
+
+Compose Files
```
Comment (mike-sul) [inline source/reference-manual/docker/compose-files.rst:3]: Maybe `Multiple Compose files` to make it in-sync with https://docs.docker.com/compose/extends/#multiple-compose-files?

File: source/reference-manual/docker/compose-files.rst [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ containers:
+  ...
+  ref_options:
+    refs/heads/devel:
+      params:
+        COMPOSE_FILES: "docker-compose.yml devel-overrides.yml"
+
+
+With this in place, Foundries.io CI will include the
+``devel-overrides.yml`` if its present when building each
```
Comment (mike-sul) [inline source/reference-manual/docker/compose-files.rst:27]:
```suggestion
``devel-overrides.yml`` if it's present when building each
```

File: source/reference-manual/docker/compose-files.rst [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+``factory-config.yml`` file in the ``ci-scripts.git`` repository.
+Here's an example of enabling an overrides file for the ``devel``
+branch::
+
+ containers:
+  ...
+  ref_options:
+    refs/heads/devel:
+      params:
+        COMPOSE_FILES: "docker-compose.yml devel-overrides.yml"
```
Comment (mike-sul) [inline source/reference-manual/docker/compose-files.rst:23]: Maybe it makes sense to extend a given example or add a new one that includes `containers::params::COMPOSE_FILES`, thus a user will know that multiple compose files may be applied to all branches.

File: source/reference-manual/docker/compose-files.rst (hunk 2 above)
Comment (doanac) [reply source/reference-manual/docker/compose-files.rst:27]: the next line completes the sentence.

**Comment by doanac** [comment]
Docs for 9a60e52e22367dc1dc8b3083ce68ea7dbea67cb3 are browesable at: https://ci.foundries.io/projects/fio-docs/builds/1238/docs/artifacts/html/index.html

**Comment by doanac** [comment]
Docs for 9a60e52e22367dc1dc8b3083ce68ea7dbea67cb3 are browesable at: https://ci.foundries.io/projects/fio-docs/builds/1239/docs/artifacts/html/index.html

File: source/reference-manual/docker/compose-files.rst (hunk 3 above)
Comment (doanac) [reply source/reference-manual/docker/compose-files.rst:23]: I thought about that some, but I think it causes problems. If you do my example with a global value like that and then merge devel->master, master apps would include that devel fragment. I think people are going to want to be explicit about this.

Comment (mike-sul) [reply source/reference-manual/docker/compose-files.rst:23]: I just considered the use-case when all branches include `common-services.yml` and `docker-compose.yml` and a specific branch may add `<branch>-overrides.yml`.

Comment (mike-sul) [reply source/reference-manual/docker/compose-files.rst:23]: https://docs.docker.com/compose/extends/#understand-the-extends-configuration

File: source/reference-manual/docker/compose-files.rst (hunk 1 above)
Comment (doanac) [reply source/reference-manual/docker/compose-files.rst:3]: fixed in force push

File: source/reference-manual/docker/compose-files.rst (hunk 3 above)
Comment (doanac) [reply source/reference-manual/docker/compose-files.rst:23]: i tried to describe this in the force-push update

**Comment by doanac** [comment]
Docs for 1d186bd8c1d6225b286bd6b19df7b42af4985bbd are browesable at: https://ci.foundries.io/projects/fio-docs/builds/1240/docs/artifacts/html/index.html

**Comment by doanac** [comment]
Docs for 7feab29f1a0abb307269ee2cbc6ac492efad02f3 are browesable at: https://ci.foundries.io/projects/fio-docs/builds/1241/docs/artifacts/html/index.html

**Review by mike-sul — APPROVED** [review]

File: source/reference-manual/docker/compose-files.rst [hunk 4]
```diff
@@ -0,0 +1,52 @@
+.. _ref-compose-files:
+
+Multiple Compose Files
+======================
+
+Docker Compose has the ability to merge multiple `Compose files`_.
+This feature can come in handy when managing a Compose file across
+different ``containers.git`` branch. Each branch may have merge
```
Comment (kprosise) [inline source/reference-manual/docker/compose-files.rst:8]: wc: "across different...branch" -> "across different branches"

**Review by kprosise — APPROVED** [review]
It looks good, with a couple small edits to make which are minor, to the point where I'm happy to patch post merge.

File: source/reference-manual/docker/compose-files.rst (hunk 4 above)
Comment (doanac) [reply source/reference-manual/docker/compose-files.rst]: https://github.com/foundriesio/docs/pull/308/commits/a8ba8fd14df2212c0d7ac54f9d3560e32069bb1e

File: source/reference-manual/docker/compose-files.rst [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+conflicts around adding a test container or settings that could be
+eliminated with multiple compose files.
+
+.. _compose files:
+   https://docs.docker.com/compose/extends/#multiple-compose-files
+
+The trick to making this work is informing the factory about the order
+that Compose files should be evaluated. This is managed by via the
```
[the remark this answers is not in the document]
Comment (doanac) [reply source/reference-manual/docker/compose-files.rst]: https://github.com/foundriesio/docs/pull/308/commits/a8ba8fd14df2212c0d7ac54f9d3560e32069bb1e

**Comment by doanac** [comment]
Docs for 7feab29f1a0abb307269ee2cbc6ac492efad02f3 are browesable at: https://ci.foundries.io/projects/fio-docs/builds/1242/docs/artifacts/html/index.html

**Review by angolini — APPROVED** [review]
LGTM

**Review by MrCry0 — APPROVED** [review]
LGTM

**Comment by doanac** [comment]
This feature has been disabled for now because of integration issues. i'm closing this and will revive at a later date

**Pull request closed without merging**